core/examples/fs_module_loader.rs

```diff
@@ -1,12 +1,88 @@
 // Copyright 2018-2023 the Deno authors. All rights reserved. MIT license.
 
+use anyhow::anyhow;
 use anyhow::Context;
 use deno_core::anyhow::Error;
+use deno_core::v8;
+use deno_core::FastString;
 use deno_core::FsModuleLoader;
 use deno_core::JsRuntime;
+use deno_core::ModuleSourceCode;
 use deno_core::RuntimeOptions;
+use std::borrow::Cow;
+use std::collections::HashMap;
 use std::rc::Rc;
 
+fn custom_module_evaluation_cb(
+  scope: &mut v8::HandleScope,
+  module_type: Cow<'_, str>,
+  module_name: &FastString,
+  code: ModuleSourceCode,
+) -> Result<v8::Global<v8::Value>, Error> {
+  match &*module_type {
+    "bytes" => Ok(bytes_module(scope, code)),
+    "text" => text_module(scope, module_name, code),
+    "url" => Ok(url_module(scope, module_name)),
```

Comment:
As I mentioned earlier, I think `type: "url"` would violate the spec and would like to remove it.

Reply:
@petamoriken, this is still work-in-progress PR and it's just an example in `deno_core`. Deno will not have support for URL imports.

Reply:
Removed now.